indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -274,11 +286,6 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
             log.makeAlert("Same Task is trying to acquire lock for overlapping interval")
                .addData("task", task.getId())
                .addData("interval", interval);
-            throw new ISE(
```

Comment:
Why did you remove the code throwing an exception here? I think it would be better to throw an exception immediately if a deadlock is found rather than waiting for the lock request to be expired.